### PR TITLE
Support for SSO basic-auth without dialogue.

### PR DESCRIPTION
**Pull request by lecler-i**
Added two new config keys : `ssoUser`, `ssoPasswordCommand` that will be used instead of the regular login/password dialogue.

Authentication will be setup with the `login` with content of `ssoUser` key, and the password will be the stdout of the execution of the command in `ssoPasswordCommand`.

Example of config :

```json
{
  "ssoUser": "Thomas",
  "ssoPasswordCommand": "gopass -o work/mycompany.com"
}
```

### CODE REVIEW

File: app/login/index.js [hunk 1]
```diff
@@ -31,12 +33,23 @@ exports.loginService = function loginService(parentWindow, callback) {
...
 
-exports.handleLoginDialogTry = function handleLoginDialogTry(window) {
+exports.handleLoginDialogTry = function handleLoginDialogTry(window, {ssoUser, ssoPasswordCommand}) {
+	console.log(ssoUser, ssoPasswordCommand);
 	window.webContents.on('login', (event, request, authInfo, callback) => {
 		event.preventDefault();
 		if (isFirstLoginTry) {
 			isFirstLoginTry = false;
-			this.loginService(window, callback);
+			if (ssoUser && ssoPasswordCommand) {
```
Comment (lecler-i) [inline app/login/index.js:41]: Maybe some better error handling here, or at least some logs if the user provided one key but not the other...

File: app/login/index.js [hunk 2]
```diff
@@ -31,12 +33,23 @@ exports.loginService = function loginService(parentWindow, callback) {
...
+				//TODO: Implement a timeout ?
+				console.log(`Retrieve password using command : ${ssoPasswordCommand}`);
+				exec(ssoPasswordCommand, (error, stdout, stderr) => {
```
Comment (lecler-i) [inline app/login/index.js]: Fairly new to nodejs, maybe document if this is running in a shell or not (to be able to use pipes ect in the provided `ssoPasswordCommand`

File: app/login/index.js (hunk 1 above)
Comment (IsmaelMartinez) [reply app/login/index.js:41]: I think this is fine for now.

File: app/config/index.js [hunk 3]
```diff
@@ -142,6 +142,18 @@ function argv(configPath, appVersion) {
...
+				default: '',
+				describe: 'User to use for SSO auth.',
+				type: 'string'
+			},
+			ssoPasswordCommand: {
+				default: '',
+				describe: 'Command to execute to retrieve password for SSO auth.',
+				type: 'string'
+			},
+
```
Comment (IsmaelMartinez) [inline app/config/index.js]: I would remove the empty lines in 145 and 156 (maybe is just the editor showing them to me)

File: app/config/index.js [hunk 4]
```diff
@@ -142,6 +142,18 @@ function argv(configPath, appVersion) {
 				describe: 'Custom Client Certs password for corporate authentication (certificate must be in pkcs12 format)',
 				type: 'string'
 			},
+
+			ssoUser: {
```
Comment (IsmaelMartinez) [inline app/config/index.js]: Could you update the 'REAME.md' file in this folder with this two options? I try to put them in alphabetical order but people tend to ignore that 😄

File: app/login/index.js (hunk 2 above)
Comment (IsmaelMartinez) [reply app/login/index.js]: You can simplify this by using execSync. Something like:

```suggestion
 try {
    const ssoPassword = execSync(ssoPasswordCommand).toString();
    callback(ssoUser, ssoPassword);
 } catch (error) {
    console.error(`Failed to execute ssoPasswordCommand. Status Code: ${error.status} with '${error.message}'`);
 }
```

I haven't tried the code but that should execute it synchronously and I think that is ok in this case.

**Review by IsmaelMartinez — COMMENTED** [review]
After applying the changes, if you can increase the number in the package.json (to 1.4.40) and add a line in the https://github.com/IsmaelMartinez/teams-for-linux/blob/develop/com.github.IsmaelMartinez.teams_for_linux.appdata.xml#L17 with what was added, then I can get it released soon (as a pre-release 1st)

**Comment by IsmaelMartinez** [comment]
Thanks for contributing!

**Comment by lecler-i** [comment]
Oh yea sorry for the  readme,config and version bump, was just a super-quick draft.
I will take your comments into consideration and do more test before a proper submit, but the change is quite simple so hopefully it won't have too much side-effects :D

**Comment by IsmaelMartinez** [comment]
sorry, wrong thumps . As it is under a feature flag (config option) I am less worried as it should not change the default use case, but I tend to release it under a pre-release so people can try it before it hits the large amount of users. Thanks again!

**Comment by IsmaelMartinez** [comment]
let me know once you are finished and I can create the pre-release. Thanks again!

**Comment by lecler-i** [comment]
Ok i've updated README, bumped the version and using `execSync`.
Codebeat is complaning about code styling tho, not sure what to do about it...

But it's good to me, tested and working.

**Comment by IsmaelMartinez** [comment]
aye, its fine to me also so merging. Thanks again! I will announce the release on the other issue.

**Comment by lecler-i** [comment]
Might wanna rename the parameters to "basicAuthLogin" / Password or smthing before release ?

**Comment by IsmaelMartinez** [comment]
done it in the 1.5.0. Thanks again for the contribution!

**Pull request merged**